ui/app/components/status-report-bar/index.ts

```diff
@@ -0,0 +1,86 @@
+import Component from '@glimmer/component';
+import { inject as service } from '@ember/service';
+import { action } from '@ember/object';
+import { tracked } from '@glimmer/tracking';
+import ApiService from 'waypoint/services/api';
+import FlashMessagesService from 'waypoint/services/flash-messages';
+import {
+  Deployment,
+  Ref,
+  ExpediteStatusReportRequest,
+  StatusReport,
+  GetJobStreamRequest,
+  GetJobStreamResponse,
+  Job,
+} from 'waypoint-pb';
+
+interface StatusReportBarArgs {
+  model: Deployment.AsObject & WithStatusReport;
+  artifactType: string;
+}
+
+interface WithStatusReport {
+  statusReport?: StatusReport.AsObject;
+}
+
+export default class StatusReportBar extends Component<StatusReportBarArgs> {
+  @service api!: ApiService;
+  @service flashMessages!: FlashMessagesService;
+  @tracked isRefreshRunning = false;
+  @tracked _statusReport?: StatusReport.AsObject;
+
+  constructor(owner: any, args: any) {
+    super(owner, args);
+    this._statusReport = this.args.model.statusReport;
```

Comment:
the `statusReport` property doesn't updated because it's nested under `model`.

You could create an intermediate getter and then use this `statusReport` in the template and the rest of the code to have it update after polling happens

```
get model() {
  return this.args.model;
}

get statusReport() {
  return this.model.statusReport;
}
```